libs/application/templates/new-primary-school/src/lib/dataSchema.ts

```diff
@@ -127,7 +127,26 @@ export const dataSchema = z.object({
     .refine((r) => r === undefined || r.length > 0, {
       params: errorMessages.siblingsRequired,
     }),
-  expectedStartDate: z.string(),
+  startingSchool: z
+    .object({
+      expectedStartDate: z.string(),
+      temporaryStay: z.string().optional(),
+      expectedEndDate: z.string().optional().nullable(),
+    })
+    .refine(({ temporaryStay }) => temporaryStay !== '', {
+      path: ['temporaryStay'],
+    })
```

Comment:
Doesn't it make more sense to add `.min(1)` and remove `refine`? 🤔 
```suggestion
  startingSchool: z
    .object({
      expectedStartDate: z.string(),
      temporaryStay: z.string().min(1).optional(),
      expectedEndDate: z.string().optional().nullable(),
    })
```

Reply:
It does. Fixed